rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -1126,14 +1126,29 @@ stream {
             {{ if not (isLocationInLocationList $location $all.Cfg.NoTLSRedirectLocations) }}
             # enforce ssl on server side
             if ($redirect_to_https) {
+                set $redirect_port {{ printf ":%v" $all.ListenPorts.HTTPS }};
```

Comment:
Why are you setting this dynamically? You can still do https://github.com/kubernetes/ingress-nginx/pull/3787/files#diff-980db9e4b88704f12338bd074839f94eL1131, no?

Reply:
fixed